include/swift/AST/GenericSignatureBuilder.h

```diff
@@ -1599,7 +1599,15 @@ class GenericSignatureBuilder::PotentialArchetype {
   /// has been resolved.
   AssociatedTypeDecl *getResolvedAssociatedType() const {
     assert(getParent() && "Not an associated type");
-    return dyn_cast<AssociatedTypeDecl>(identifier.assocTypeOrConcrete);
+
+    auto *assocTypeOrConcrete = identifier.assocTypeOrConcrete;
+    if (auto *alias = dyn_cast<TypeAliasDecl>(assocTypeOrConcrete)) {
+      auto type = alias->getUnderlyingTypeLoc().getType();
+      if (auto *DM = type->getAs<DependentMemberType>())
+        return DM->getAssocType();
```

Comment:
This won't work, because the underlying type isn't necessarily a dependent member type with the same `Self`. The callers will need to substitute into the `TypeAliasDecl` to determine which type it refers to; it

Reply:
Ok, I'll try to figure it out, although I have no clue what any of the words you just wrote mean :)